apps/rush-lib/src/logic/PublishUtilities.ts

```diff
@@ -642,25 +642,32 @@ export class PublishUtilities {
       dependencies[change.packageName] &&
       !PublishUtilities._isCyclicDependency(allPackages, parentPackageName, change.packageName)
     ) {
-      const requiredVersion: string = new DependencySpecifier(
+      const requiredVersion: DependencySpecifier = new DependencySpecifier(
         change.packageName,
         dependencies[change.packageName]
-      ).versionSpecifier;
+      );
+      const isAnyWorkspaceVersion: boolean =
+        requiredVersion.specifierType === DependencySpecifierType.Workspace &&
+        requiredVersion.versionSpecifier === '*';
       const alwaysUpdate: boolean =
-        !!prereleaseToken && prereleaseToken.hasValue && !allChanges.hasOwnProperty(parentPackageName);
+        (!!prereleaseToken && prereleaseToken.hasValue && !allChanges.hasOwnProperty(parentPackageName)) ||
+        isAnyWorkspaceVersion;
 
       // If the version range exists and has not yet been updated to this version, update it.
-      if (requiredVersion !== change.newRangeDependency || alwaysUpdate) {
+      if (requiredVersion.versionSpecifier !== change.newRangeDependency || alwaysUpdate) {
         let changeType: ChangeType;
         // Propagate hotfix changes to dependencies
         if (change.changeType === ChangeType.hotfix) {
           changeType = ChangeType.hotfix;
         } else {
           // Either it already satisfies the new version, or doesn't.
           // If not, the downstream dep needs to be republished.
-          changeType = semver.satisfies(change.newVersion!, requiredVersion)
-            ? ChangeType.dependency
-            : ChangeType.patch;
+          // The downstream dep will also need to be republished if using `workspace:*` as this will publish
+          // as the exact version.
+          changeType =
+            semver.satisfies(change.newVersion!, requiredVersion.versionSpecifier) && !isAnyWorkspaceVersion
```

Comment:
Seems like this should be

```suggestion
            isAnyWorkspaceVersion || semver.satisfies(change.newVersion!, requiredVersion.versionSpecifier)
```

so a `workspace:*` version gets a "dependency"-type bump. Isn't that what `ChangeType.dependency` is for?

Reply:
No, not from my understanding. A `ChangeType.dependency` bump is used to signify that a dependency was bumped, but that the current version range can satisfy the new dependency. That change type doesn't actually bump the version, but patch does.